src/overlays/actors/ovl_Boss_03/z_boss_03.c

```diff
@@ -4,31 +4,280 @@
  * Description: Gyorg
  */
 
+/**
+ * Some notes:
+ *
+ * Global flags:
+ * - gSaveContext.eventInf[5] & 0x40: Enabled when Gyorg's intro cutscene has been watched
+ * - gSaveContext.weekEventReg[55] & 0x80: Checked to know if Gyorg has been defeated
+ *
+ * Seaweed:
+ * - Refers to the seaweed at the bottom of the stage where Player fights Gyorg
+ * - The default Gyorg actor will spawn 5 other Gyorg instances using the parameter GYORG_PARAM_SEAWEED to spawn them
+ * - Seaweed can bend when then main Gyorg instance (the actual boss) or Player are near
+ *
+ * This actor makes some heavy assumptions about the positions of the room where it is spawned
+ *
+ * Cutscenes:
+ * - There are 3 cutscenes:
+ *   - IntroCutscene: The cs which is played when Player falls into the main room. It also shows Gyorg's titlecard
+ *   - SpawnSmallFishesCutscene: The short cs which is played when Gyorg is spawning the small fishes (EnTanron3). This
+ * is triggered when Gyorg reaches half of its life
+ *   - DeathCutscene: Played when Gyorg dies. Showing it splashing and becoming smaller each time until it disappears
+ * - This actor mainly handles the 3 cutscenes it has manually (instead of relying on existing systems for it)
+ *
+ * Main behaviour:
+ * - Gyorg has two types of behaviour depending on Player's state:
+ *   - If Player is standing on the main platform, then Gyorg follows the PrepareCharge -> Charge -> JumpOverPlatform
+ * branch
+ *   - Otherwise, Gyorg follows the ChasePlayer -> CatchPlayer -> ChewPlayer
```

Comment:
Probably want to add "branch" to the end; the current sentence makes me ask "the what?"
```suggestion
 *   - Otherwise, Gyorg follows the ChasePlayer -> CatchPlayer -> ChewPlayer branch
```